### PR TITLE
Fix tabular headers

### LINKED ISSUE #17506: Tabular layouts freeze when adding a datetime column
### Describe the Bug

Whenever I try to add a `datetime` column to the tabular layout, it freezes things up such that I eventually have to reload the app.

I don't know why it freezes up exactly, because on 22.4 it doesn't even load the column values and just totally freezes up, while on 22.0 it does add the column, but there are errors in the console and eventually when trying to navigate back to the page in question, it hangs forever.....

BUT I do know the the cause: the `datetime` display is getting an empty array as a value, which it doesn't know how to handle. It's expecting null or a string. 

The culprit is this line: https://github.com/directus/directus/blob/1e824f7f2184b9c1c95bf8cee1f2b8a67111171e/app/src/layouts/tabular/tabular.vue#L296
For some godforsaken reason that returns `true` for an empty array, meaning when values aren't yet retrieved from the backend but the fields have changed and the table is eagerly trying to load them, this will return an empty array to the `render-display` element which causes this small bit of havoc. I added an empty array check on `matchingValues` and everything works smoothly now. 

### To Reproduce

I don't think you need anything fancy here: just use a tabular layout and use the + button to add a datetime column to it. 

### Errors Shown

```
RangeError: Invalid time value
    at format (chunk-SVH7MBRM.js?v=95da87c8:1040:11)
    at localizedFormat (localized-format.ts?t=1676332598802:4:10)
    at watch.immediate (datetime.vue?t=1676332598802:76:32)
```

```
Cannot read properties of null (reading 'emitsOptions')
...then a bunch of chunk files
```

### What version of Directus are you using?

22.4

### What version of Node.js are you using?

16.14.2

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### How are you deploying Directus?

locally, Docker

### LINKED ISSUE #17670: 9.23 - O2M sub-columns not displaying in table layout
### Checklist

- [X] I'm using the [latest version of Directus](https://github.com/directus/directus/releases)
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes the problem.
- [X] I've completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).

### Describe the Bug

Reproduction SQLite3 Database:
[test.db.zip](https://github.com/directus/directus/files/10864556/test.db.zip)
Username: `test@test.com`
Password: `test`

Tested With:
- Integer primary keys
- UUID primary keys
- Custom primary keys
- SQLite3

Not Tested:
- Other databases
- Other relationship types
- Other layout & preset scenarios

Likely introduced in: https://github.com/directus/directus/pull/15269

In the table layout, columns selected from the child table of an O2M relationship do not display at all and seem to conflict with O2M field on the parent table.

Sub-columns `Title` and `Path` invisible without the parent O2M column selected:
image

Sub-columns `Title` and `Path` display the primary key of the child table when the parent O2M column is selected:
image

There are a slew of other interaction/UI bugs relating to this, but they're a bit hard to describe. 

### To Reproduce

1. Create a table to be on the "Many" end of an O2M relationship. Add a couple input columns to it.
image

2. Create another table to be on the "One" end of the O2M relationship. Add an input column to it (just for demonstration), and an O2M relation to the "Many" table.
image

3. Create several rows in the "One" table, each relating to a unique row in the "Many" table.
image

4. In the table view for the "One" table, add the input columns from the "Many" table to the table headers:
image

5. Note that the related fields do not display. Clicking on the "jump to item" links will navigate to an `content/TABLE/undefined` route.

6. Adding the O2M column directly to the headers results in all sub-columns displaying the primary key for the O2M field, and rearranging them causes the names all to review to the field name for the O2M field.
image

### Hosting Strategy

Self-Hosted (Custom)

### LINKED ISSUE #17716: 9.23 - Displaying of relational fields of Translations relation is broken in Tabular layout
### Checklist

- [X] I'm using the [latest version of Directus](https://github.com/directus/directus/releases)
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes the problem.
- [X] I've completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).

### Describe the Bug

9.23 - Displaying of relational fields of Translations relation is broken in Tabular layout.

https://user-images.githubusercontent.com/26179694/223399302-b656f256-de13-41ff-813c-8b627733c992.mp4

This only started happening since version 9.23.
I suggest looking in the direction of Discussion #16508 and PR #16507. I think this display algorithm will be the most acceptable.

### To Reproduce

1. Create collection A with Translations relation (to collection A_translations).
2. Add some items.
3. Go to Content module.
4. Try to add `Translations -> Languages Code` column. It will show the empty values.
5. Then try to add `Translations` column. It will show the data within Translations display in this column correctly. The `Translations -> Languages Code` will show the same Translations display with the same data (not a Languages Code field only).
6. The try to move one of these columns (d'n'd) to another place. After this action, all both `Translations` and `Translations -> Languages Code` columns will have the same name — `Translations`.
7. Then try to delete one of these columns. Both columns will be removed.

### Hosting Strategy

Self-Hosted (Docker Image)

### PR DESCRIPTION
**Pull request by Nitwel**
## Description

This PR reworks the way the tabular layout works with aliased requests, as the old way was really hacky by trying to merge aliases depending on their originally name. The implementation now only ever passes the the data a certain column had requested back to it so there is no collision in the data each display requires.

And the other thing that got improved is that now only nested fields of translations are rendered in a special way. 

Fixes #17716
Fixes #17506
Fixes #17670
Fixes ENG-690

### CODE REVIEW

**Comment by paescuj** [comment]
Is this also going to address #17506 (reopened because of the #17816)?

**Review by paescuj — APPROVED** [review]
Only a small comment, otherwise looks great 😃👍

**Review by paescuj — CHANGES_REQUESTED** [review]
**Small addition:**

The value passed to `render-display` (computed in `getDisplayValue`) is now an array in case of relational fields. While this doesn't seem to be an issue, the following warning is shown in the console:

```
[Vue warn]: Invalid prop: type check failed for prop "value". Expected String | Number, got Array  
  at <FormattedValue interface="input" interface-options=null value= 
Array [ "test" ]
  ... > 
  at <VErrorBoundary key=2 name="display-formatted-value" > 
  at <RenderDisplay value= 
Array [ "test" ]
 display="formatted-value" options=null  ... > 
  at <TableRow headers= 
Array(3) [ {…}, {…}, {…} ]
 item= 
Object { category: "category1", id: 1, 1be40: (1) […], 5622e2d6: (1) […] }
 show-select="multiple"  ... > 
  at <Draggable key=3 modelValue= 
Array [ {…} ]
 onUpdate:modelValue=fn  ... > 
  at <VTable key=0 ref="table" modelValue= 
Array []
  ... > 
  at <Tabular key=1 class="layout" tableHeaders= 
Array(3) [ {…}, {…}, {…} ]
  ... > 
  at <PrivateView key=1 title="Parent" small-header=undefined > 
  at <AsyncComponentWrapper key=1 title="Parent" small-header=undefined > 
  at <TabularWrapper ref="layoutRef" selection= 
Array []
 onUpdate:selection=fn  ... > 
  at <ContentCollection collection="parent" bookmark=null archive=null > 
  at <CollectionOrItem collection="parent" bookmark=undefined archive=null  ... > 
  at <RouterView> 
  at <RouterPassthrough onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< 
#text ""
 > > 
  at <RouterView> 
  at <RouterPassthrough onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< 
#text ""
 > > 
  at <RouterView key=1 > 
  at <App> chunk-2GTISQ3F.js:1641:13
```

So we can probably just extend
https://github.com/directus/directus/blob/46c1666607abef720d2fb668c33e80e041ee4dc5/app/src/displays/formatted-value/formatted-value.vue#L37
by `Array`, correct?

**Comment by Nitwel** [comment]
Added Array to the formatted-value display. 👍

File: app/src/layouts/tabular/tabular.vue [hunk 1]
```diff
@@ -280,30 +275,25 @@ const showManualSort = computed(() => {
...
+		} else {
+			result[key] = item[key];
+		}
+		return result;
+	}, {});
 
-	// merge into a single record
-	const result = matchingValues.reduce((result, data) => merge(result, data), {});
-	return !isEmpty(result) ? result : null;
+	return get(unAliasedItem, key);
```
[the remark this answers is not in the document]
Comment (paescuj) [reply app/src/layouts/tabular/tabular.vue]: `dealiased` is even better 😄🫶

**Review by paescuj — APPROVED** [review]

**Review by rijkvanzanten — APPROVED** [review]

**Comment by hanneskuettner** [comment]
This PR caused the broken calendar layout display template #18206 and will also break the CSV export as it changed the return type (and function as far as I can tell) of `useAliasFields`.

In both cases the expected key is `aliasField` which has been removed.

See https://github.com/directus/directus/blob/42e19633456341b1cae4e19db7e52ac9e25bd08b/app/src/utils/render-string-template.ts#L70 

and 
https://github.com/directus/directus/blob/42e19633456341b1cae4e19db7e52ac9e25bd08b/app/src/utils/save-as-csv.ts#L22

**Pull request merged**